agnpy/tests/test_spectra.py

```diff
@@ -86,6 +93,76 @@ def broken_power_law_times_gamma_integral(k_e, p1, p2, gamma_b, gamma_min, gamma
     return k_e * (term_1 + term_2)
 
 
+# Data are generated, following the four already implemented distributions of agnpy.
+# The interpolation function is derived for each data set and then is interpolated and
+# data points are being compared:
+
+def pwl_data(k_e_test,p_test,gamma_min_test,gamma_max_test):
+
+    pwl_data = np.zeros((2,100),float)
+    gamma1 = np.logspace(np.log10(gamma_min_test),np.log10(gamma_max_test),100)
+    pwl_data[0,:] = gamma1
```

Comment:
Why are you creating this matrix of 2 * 100?
Can't you pass directly the array (which is always the same, `gamma1`) and then the values generated by `pwl_test(gamma1)` - or `bpwl_test(gamma1)` - to your function?